src/OrleansRuntime/MultiClusterNetwork/MultiClusterData.cs

```diff
@@ -0,0 +1,303 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Orleans.MultiCluster;
+
+namespace Orleans.Runtime.MultiClusterNetwork
+{
+    /// <summary>
+    /// Data stored and transmitted in the multicluster network. 
+    /// IMPORTANT: these objects can represent full state, partial state, or delta.
+    /// So far includes multicluster-configuration and multicluster-gateway information.
+    /// Data is gossip-able.
+    /// </summary>
+    [Serializable]
+    public class MultiClusterData : IEquatable<MultiClusterData>, IMultiClusterGossipData
```

Comment:
MultiClusterData is public because it appears in the public IGossipChannel interface.
The latter is public because it is an extension point: we are going to have alternative channel providers that do not depend on Azure storage.